cunumeric/module.py

```diff
@@ -1460,17 +1460,30 @@ def check_shape_dtype(
             f"All arguments to {func_name} "
             "must have the same number of dimensions"
         )
-    if ndim > 1 and _builtin_any(
-        shape[1:axis] != inp.shape[1:axis]
-        and shape[axis + 1 :] != inp.shape[axis + 1 :]
-        for inp in inputs
-    ):
-        raise ValueError(
-            f"All arguments to {func_name} "
-            "must have the same "
-            "dimension size in all dimensions "
-            "except the target axis"
-        )
+
+    if func_name == "stack":
+        shapes = {inp.shape for inp in inputs}
+        if len(shapes) != 1:
+            raise ValueError(
+                f"all input arrays must have the same shape for {func_name}"
+            )
+    else:
+        if ndim == 1 and func_name == "hstack":
+            axis = 0
+        if ndim >= 1:
+            if axis < 0 and axis >= -ndim:
+                axis += ndim
```

Comment:
I think `normalize_axis_index` can be used here

Reply:
We don't use ```normalize_axis_index``` here because it raises error for the column_stack case where axis=1 and ndim=1.

Reply:
After function split, we can use  normalize_axis_index here now as I have enhanced the column_stack case for ndim=1.